### ISSUE TITLE
[ingress/controllers/nginx] ip whitelisting

### ISSUE THREAD

**Issue by sander-su** [kubernetes/contrib#1141]
We run our kubernetes cluster in aws but would like to restrict access to some services based on ip. This could be users coming from our corporate network our devs who use a vpn to get into the aws vpc but do not need a fully kubernetes account. Basic auth is a very nice step forward but not restrictive enough yet.

I would be great if we define comma seperated cidr blocks also via a configmap.
e.g. internal would be 10.0.0.0/16 for aws vpc
corporate would be x.x.x.x/x, y.y.y.y/y
Using a configmaps gives the option of changing (corporate) ip's at one place instead of every ingress.

**Comment by aledbf** [comment]
@sander-su noted for next version. I will add a global option (in the configmap) and an annotation in the ingress.

**Comment by bprashanth** [comment]
Do we actually get an accurate source ip using `hostPort` or do we need to run the controller with net=host?

**Comment by eparis** [comment]
I thought on modern docker (like 1.8+ 1.9+) they used an iptables forwarding implementation for hostPort which left the src IP intact...

**Comment by sander-su** [comment]
We run as daemonset using hostPort, I'll check tomorrow regarding IP's in logging.

You triggered me however:
It depends however what kind of loadbalancer is used what source could be used for filtering.
AWS ELB can do Proxy protocol
http://docs.aws.amazon.com/ElasticLoadBalancing/latest/DeveloperGuide/enable-proxy-protocol.html 
no experience with that however, believe I saw a config for this in the ingress. 
Other option would be to look for x-forwarded-headers.

So a configuration item is needed to select one of those

**Comment by bprashanth** [comment]
Yeah you can definitely use proxy protocol. As @eparis said docker does have an iptables mode for host port but it wasn't the default in 1.9, not sure about 1.11 which is the version we ship with in kube-1.3. That said, kube-1.3 will use kubenet (CNI network plugin) by default, and that *does* handle host port via iptables, by default. So to answer the srcip hostport question one needs to builds a docker x kube version matrix, apparently.

**Comment by sander-su** [comment]
On my setup:
kubernetes 1.2.4 / coreos 1010.4.0
AWS ELB set to proxy protocol
Ingres using hostPort
Configmap for ingres:
  use-proxy-protocol: "true"
  proxy-real-ip-cidr: "172.16.0.0/12"
I get my correct source ip logged.
So looks like net=host (or hostNetwork: true) is not needed when using headers (proxy protocol or x-forward)
Need to verify if the ip that is logged as source ip is also used for whitelisting, or what config this requires.

I used the overlay network cidr because I did not use net=host and the docker0 address is otherwise used as source, which is in the overlay range.

When I tried hostNetwork: true it was bugging me that port 10249 (health) was already in use and the ingress would not start (hyperkube proxy was using that and was already running net=host, so conflict here). Is there an option to specify this port? (could not find it)
